### PR TITLE
Side box

### PR DESCRIPTION
**Pull request by shaobinspire**

### CODE REVIEW

**Review by spireken — APPROVED** [review]
Looks good.

**Comment by kranar** [comment]
API was not submitted for review and would not have been approved had it been.

File: Applications/Spire/Source/Ui/SideBox.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#include "Spire/Ui/SideBox.hpp"
+#include "Spire/Spire/LocalValueModel.hpp"
+#include "Spire/Ui/ArrayListModel.hpp"
+
+using namespace Nexus;
+using namespace Spire;
+
+SideBox* Spire::make_side_box(QWidget* parent) {
+  return make_side_box(Side::NONE, parent);
```
Comment (kranar) [inline Applications/Spire/Source/Ui/SideBox.cpp:9]: ```Side::NONE``` is not a valid case, the components.xd specifies that only Buy and Sell are valid.

File: Applications/Spire/Source/Ui/SideBox.cpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+SideBox* Spire::make_side_box(Side current, QWidget* parent) {
+  return make_side_box(std::make_shared<LocalSideModel>(current), parent);
+}
+
+SideBox* Spire::make_side_box(std::shared_ptr<SideModel> model,
+    QWidget* parent) {
+  auto settings = SideBox::Settings();
+  auto cases = std::make_shared<ArrayListModel>();
+  cases->push(Side(Side::ASK));
```
Comment (kranar) [inline Applications/Spire/Source/Ui/SideBox.cpp:20]: Spec orders things buy and sell.

File: Applications/Spire/Source/Ui/SideBox.cpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  return make_side_box(Side::NONE, parent);
+}
+
+SideBox* Spire::make_side_box(Side current, QWidget* parent) {
+  return make_side_box(std::make_shared<LocalSideModel>(current), parent);
+}
+
+SideBox* Spire::make_side_box(std::shared_ptr<SideModel> model,
+    QWidget* parent) {
+  auto settings = SideBox::Settings();
```
Comment (kranar) [inline Applications/Spire/Source/Ui/SideBox.cpp:18]: Avoid constantly doing the same work over and over.

File: Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp [hunk 4]
```diff
@@ -1660,6 +1688,17 @@ UiProfile Spire::make_security_list_item_profile() {
   return profile;
 }
 
+UiProfile Spire::make_side_box_profile() {
+  auto properties = std::vector<std::shared_ptr<UiProperty>>();
+  populate_widget_properties(properties);
+  auto current_property = define_enum<Side>(
+    {{"Buy", Side::ASK}, {"Sell", Side::BID}});
```
Comment (kranar) [inline Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp:1695]: Mismatching values.

**Review by kranar — CHANGES_REQUESTED** [review]

File: Applications/Spire/Source/Ui/SideBox.cpp [hunk 5]
```diff
@@ -5,8 +5,21 @@
 using namespace Nexus;
 using namespace Spire;
 
+namespace {
+  SideBox::Settings setup() {
+    static auto settings = SideBox::Settings();
```
Comment (kranar) [inline Applications/Spire/Source/Ui/SideBox.cpp:10]: Use IIFE.

**Review by kranar — CHANGES_REQUESTED** [review]

**Pull request merged**